### PR TITLE
Guidelines for incomplete stubs

### LINKED ISSUE #2648: Incomplete stubs
When stubs for new packages are submitted, they are often incomplete. I think that's okay, we can't contributors to annotate a complete, complex package when they use only a small part of it. Nevertheless these incomplete packages have value.

But they have one problem: Correct code that previously type checked with `--ignore-missing-imports` or similar flags will now flag errors if the new stub is incomplete. Similar, new code needs to be littered with `# type: ignore`s.

I suggest to add something like this to `CONTRIBUTING`:

* There must be a stub for each sub-module.
* Incomplete or empty modules must contain `__getattr__`, with a `# incomplete` comment.
* Same for incomplete classes.
* Functions and methods must have all arguments, but they can be left unannotated.

### PR DESCRIPTION
**Pull request by srittau**
Closes #2648

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -145,6 +145,38 @@ Example:
 def list2cmdline(seq: Sequence[str]) -> str: ...  # undocumented
 ```
 
+### Incomplete stubs
+
+We accept partical stubs, especially for larger packages. These need to
```
Comment (utkarsh2102) [inline CONTRIBUTING.md]: `partical`?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -145,6 +145,38 @@ Example:
...
+  A better alternative is to create empty stubs for all sub-modules and
+  mark them as incomplete individually.
+
+Example of a partial module with a partial class `Foo` and a partially
+annotated function `bar()`:
+
+```python
+def __getattr__(name: str) -> Any: ...  # incomplete
+
+def Foo:
```
[the remark this answers is not in the document]
Comment (srittau) [reply CONTRIBUTING.md]: I have started to introduce this typo a lot in my code lately. Fortunately PyCharm catches it. Will fix.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -145,6 +145,38 @@ Example:
 def list2cmdline(seq: Sequence[str]) -> str: ...  # undocumented
 ```
 
+### Incomplete stubs
+
+We accept partial stubs, especially for larger packages. These need to
+follow the following guidelines:
+
+* Included functions and methods must list all arguments, but the arguments
+  can be left unannotated. Do not use `Any` to mark unannotated arguments
```
[the remark this answers is not in the document]
Comment (srittau) [reply CONTRIBUTING.md]: I feel leaving it unannotated is better to mark it as such and use `Any` for things that can't be annotated any better (at least at the moment).

[the remark this answers is not in the document]
Comment (srittau) [reply CONTRIBUTING.md]: I have merged and removed that item from the `Any` guidelines. Could you check whether that is ok?

**Pull request merged**